finviz/helper_functions/request_functions.py

```diff
@@ -6,49 +6,72 @@
 
 urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
 
+CONCURRENT_CONNECTIONS = 20
+CONNECTION_TIMEOUT = 3000
```

Comment:
This is a bit ambiguous. Why only 20 concurrent connections and not 30 let's say? My point is, what is your reasoning behind this number? Same for line 10 - why 3 seconds connection timeout and not 30 or 60 which is standard in waiting for a response?

Reply:
I also believe that if we add this feature there should be a way to modify these two variables. What if the finviz servers are overwhelmed and the page loads after 10 seconds and not 3? The user will never get a successful output. There should be some kind of a warning or at least it should be clearly stated in the documentation that the standard is 20 concurrent connection and 3 secs timeout.

Reply:
> 
> 
> This is a bit ambiguous. Why only 20 concurrent connections and not 30 let's say? My point is, what is your reasoning behind this number? Same for line 10 - why 3 seconds connection timeout and not 30 or 60 which is standard in waiting for a response?

In my tests on a fast connection, with 30 concurrent connections it did timeout on some whereas with 20 it seemed to be stable, feel free to change.

Reply:
> 
> 
> I also believe that if we add this feature there should be a way to modify these two variables. What if the finviz servers are overwhelmed and the page loads after 10 seconds and not 3? The user will never get a successful output. There should be some kind of a warning or at least it should be clearly stated in the documentation that the standard is 20 concurrent connection and 3 secs timeout.

Absolutely, it could well be a variable, perhaps time to create `variables.py` and/or pass it as a parameter to the function?

Reply:
We can make it a default parameter. However, if you think about it, the user has no access to the `http_request_get` method whatsoever. The interface is the Screener, and I don't think its adequate to add concurrent connection as a parameter for Screener. The better option will be to create `variables.py` to store global variables somehow. You have to think about that.